tests/adapter/dbt/tests/adapter/basic/test_adapter_methods.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+
+from dbt.tests.util import run_dbt
+from dbt.tests.fixtures.project import write_project_files
+from dbt.tests.tables import TableComparison
+
+
+tests__get_columns_in_relation_sql = """
+{% set columns = adapter.get_columns_in_relation(ref('model')) %}
+{% set limit_query = 0 %}
+{% if (columns | length) == 0 %}
+    {% set limit_query = 1 %}
+{% endif %}
+
+select 1 as id limit {{ limit_query }}
+
+"""
+
+models__upstream_sql = """
+select 1 as id
+
+"""
+
+models__expected_sql = """
+-- make sure this runs after 'model'
+-- {{ ref('model') }}
+select 2 as id
+
+"""
+
+models__model_sql = """
+
+{% set upstream = ref('upstream') %}
+
+{% if execute %}
+    {# don't ever do any of this #}
+    {%- do adapter.drop_schema(upstream) -%}
+    {% set existing = adapter.get_relation(upstream.database, upstream.schema, upstream.identifier) %}
+    {% if existing is not none %}
+        {% do exceptions.raise_compiler_error('expected ' ~ ' to not exist, but it did') %}
+    {% endif %}
+
+    {%- do adapter.create_schema(upstream) -%}
+
+    {% set sql = create_view_as(upstream, 'select 2 as id') %}
+    {% do run_query(sql) %}
+{% endif %}
+
+
+select * from {{ upstream }}
+
+"""
+
+
+class BaseCaching:
+    @pytest.fixture(scope="class")
+    def tests(self):
+        return {"get_columns_in_relation.sql": tests__get_columns_in_relation_sql}
+
+    @pytest.fixture(scope="class")
+    def models(self):
+        return {
+            "upstream.sql": models__upstream_sql,
+            "expected.sql": models__expected_sql,
+            "model.sql": models__model_sql,
+        }
+
+    @pytest.fixture(scope="class")
+    def project_files(
+        self,
+        project_root,
+        tests,
+        models,
+    ):
+        write_project_files(project_root, "tests", tests)
+        write_project_files(project_root, "models", models)
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {
+            "name": "adapter_methods",
+        }
+
+    # snowflake need all tables in CAP name
+    @pytest.fixture(scope="class")
+    def equal_tables(self):
+        return ["model", "expected"]
+
+    def test_adapter_methods(self, project, equal_tables):
+        run_dbt(["compile"])  # trigger any compile-time issues
```

Comment:
@gshank  This works for snowflake, redshift, bigquery and here, it just doesn't work for spark(any type of connection method). the error got thrown was `Table or view not found: test16486637045872746973_test_basic.model`. Do you know what the issue could be?

Reply:
Did you set the 'file_format' for models?

Reply:
The 'basic' tests worked on spark, and I would think if there was an issue it would have shown up there. I notice that the 'adapter_test_methods' hasn't been copied to dbt-spark. Maybe there's a reason?

Reply:
Short story this doesn't work because when we do `dbt compile` we would actually try to run the macro defined [here](https://github.com/dbt-labs/dbt-core/pull/4939/files#diff-a98f96e4919c93a28aee46c131ac9f52bad92403794d7ed03a947c73cb24a96fR9), but in `dbt run` we don't since it is in a test. What's making spark different from others is that when describing a table that doesn't exist, the spark will throw an error and others will return an empty list. Here's the [internal thread](https://dbt-labs.slack.com/archives/C0131TY7EEA/p1649170666282589) about this issue